### ISSUE TITLE
Add config option to identify lodash object

### ISSUE THREAD

**Issue by tauren** [jjt/grunt-lodash-autobuild#1]
This grunt task works great! However, it assumes that the lodash object is always `_`, but this may vary across all of the modules within a project. In other words, I might have the following AMD module, which is not currently supported by grunt-lodash-autobuild:

``` javascript
define(['lodash'], function(lodash) {
  lodash.uniq(['foo','bar','baz','bar']);
});
```

To use the grunt task, I would have to refactor my code to this:

``` javascript
define(['lodash'], function(_) {
  _.uniq(['foo','bar','baz','bar']);
});
```

Furthermore, in a team with many developers, the internal variable name used might vary between modules. Some developers might use `_` while others use `lodash` or even `underscore`.

I recommend that a configuration option be added to allow support for one or more different lodash object names. From a quick review of the code, this isn't as simple as an option to override the `/_\.\w*/gi` regex, since the following line also has `_.` hardcoded:
https://github.com/jjt/grunt-lodash-autobuild/blob/master/tasks/lodash_autobuild.js#L32

**Comment by jjt** [comment]
Thanks! And I appreciate the suggestion too. A config option is the first thing I thought of as well (before I got to the bottom of your comment) and would be the way to handle it, as you say.

```
    lodashAutobuild: {
      src: ['src/js/**/*.js', 'src/coffee/**/*.coffee'],
      options: {
        lodashObjects: [ '_', 'underscore', 'lodash' ]
      }
    }
```

**Comment by jjt** [comment]
Just updated. Should work as described, defaults to `[ '_' ]`.

Closed by https://github.com/jjt/grunt-lodash-autobuild/commit/8620a4a2b33176a649e2573a1f0c5f584c81fd69

**Comment by tauren** [comment]
Thanks for the quick fix! I want to point out an issue I discovered. Consider this file:

``` javascript
require.config({
  paths: {
    lodash    : '../vendor/lodash.build',
    domReady  : '../vendor/domReady'
  }
});
```

Running the grunt task with `lodashObjects: [ '_', 'underscore', 'lodash' ]` outputs this:

```
Running "lodashAutobuild:src" (lodashAutobuild) task
Found the following lodash functions:
build, forEach, forEach, isEmpty, keys
```

Notice _build_ is in there?  Ideally it wouldn't be there. I can live with it for now, but a more sophisticated regex might be nice to have eventually that can exclude instances within strings. 

Personally, I would consider moving away from using a regex and instead use an AST parser to identify actual variable names, filter for only the variable names specified in `lodashObjects`, and then add all of the methods called on those variables. You could look at http://esprima.org/doc/index.html#ast or http://lisperator.net/uglifyjs/ast for more information.

**Comment by jjt** [comment]
Hmm, right. And it appears that grunt-lodash barfs if it finds an unknown lodash method in the list.

Yeah, I might have to move it to an AST-based solution. I'll at least have to construct a list of the lodash API methods to filter the ones we found.

Thanks again for the feedback.

**Comment by jjt** [comment]
Okay, fixed in 2a87612.

**Comment by tauren** [comment]
Thanks!
